csaps/_sspumv.py

```diff
@@ -196,6 +223,33 @@ def spline(self) -> SplinePPForm:
         """
         return self._spline
 
+    def compute_gcv(self, y, y_pred) -> float:
+        """Computes the GCV using the degrees of freedom, which must be computed upon spline creation
+
+        Parameters
+        ----------
+
+        y : 1-d array-like
+            Points the spline was evaluated at
+
+        y_pred : 1-d array-like
+            Predicted values returned from the generated spline object
+
+        Returns
+        -------
+        gcv : float
+            The generalized cross validation criterion
+
+
+        """
+        if not self.degrees_of_freedom:
+            raise ValueError("You must recreate the spline with the calculate_degrees_of_freedom parameter set to True")
+
+        y = np.asarray(y, dtype=np.float64)
+        y_pred = np.asarray(y_pred, dtype=np.float64)
+        self.gcv: float = np.linalg.norm(y-y_pred)**2 / (y.size - self.degrees_of_freedom)**2
```

Comment:
Here we use y prediction values from the evaluated spline result as I understand. How about size(y) != size(y_pred)? We can evaluate the spline on an arbitrary number of points and the length of input vector will not equal to the vector of the evaluated spline values.